### PR TITLE
fix: update insecure tar dependency

### PR DESCRIPTION
**Pull request by luca-gr4vy**
**Description:** Adds resolution to tar `^6.2.1`, fixing the vulnerability related to it https://github.com/gr4vy/gr4vy-react-native/security/dependabot/53. Also ignores a few vulnerabilities found by trivy that are not important and can only be fixed by upgrading the project.

**Note:** To test I had to downgrade to Xcode 14.3 (we originally set up the project with that version), as working with Xcode 15 would fail to launch the example app. We'd need to upgrade the React Native version, I created a ticket for it https://gr4vy.atlassian.net/browse/TA-6984

### CODE REVIEW

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -25,3 +25,5 @@ runs:
         yarn install --cwd example --frozen-lockfile
         yarn install --frozen-lockfile
       shell: bash
+      env:
+        SKIP_YARN_COREPACK_CHECK: '1'
```
Comment (luca-gr4vy) [inline .github/actions/setup/action.yml:29]: Skipping corepack for these tasks due to this https://github.com/gr4vy/gr4vy-react-native/actions/runs/8982545520/job/24670387490?pr=26#step:3:46. We don't need to use the same exact yarn version for linting, but will probably upgrade the supported one in the future.

**Comment by gryevns** [comment]
Do we have a way of testing this e2e?

**Comment by luca-gr4vy** [comment]
> Do we have a way of testing this e2e?

Not really, we don't have it for react native yet as it's notoriously complicated to set up

**Review by gryevns — APPROVED** [review]

**Comment by gr4vy-code** [comment]
:rocket: PR was released in [`v1.1.0`](https://github.com/gr4vy/gr4vy-react-native/releases/tag/v1.1.0) :rocket:

**Pull request merged**